libraries/Microsoft.Bot.Builder/DictionaryExtensions.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder
+{
+    /// <summary>
+    /// Extension method on object <see cref="object"/>.
+    /// </summary>
+    public static partial class DictionaryExtensions
+    {
+        private static JsonSerializer _coerceSerializer = new JsonSerializer() { TypeNameHandling = TypeNameHandling.None };
+
+        /// <summary>
+        /// Extension Method on object to coerce the value of the dictionary to type T.
+        /// </summary>
+        /// <remarks>This allows the dictionaries which are string => JToken to be able to affinitize to a typed instance.</remarks>
+        /// <param name="dict">object to cast.</param>
+        /// <param name="property">property name.</param>
+        /// <typeparam name="T">type to which object should be casted.</typeparam>
+        /// <returns>T.</returns>
+        public static T MapValueTo<T>(this IDictionary<string, object> dict, string property)
+        {
+            if (dict.TryMapValueTo(property, out T result))
+            {
+                return result;
+            }
+
+            return default(T);
+        }
+
+        /// <summary>
+        /// Extension Method on IDictionary&lt;string,object&gt; to coerce the value of the dictionary to type T.
+        /// </summary>
+        /// <remarks>This allows the dictionaries which are string => JToken to be able to affinitize to a typed instance.</remarks>
```

Comment:
Q: This returns a copy of the original object, not the object itself, right? (I haven't played with the different options so not sure and it may be worth documenting)

Reply:
Clarified the behavior in remarks.